src/dependency.rs

```diff
@@ -26,6 +26,9 @@ pub struct Dependency {
     /// If the dependency is renamed, this is the new name for the dependency
     /// as a string.  None if it is not renamed.
     rename: Option<String>,
+
+    /// Features that are exposed by the dependency
```

Comment:
I'm a bit mixed on us using `Dependency` to do more than just represent the dependency field but to track additional metadata.

Reply:
I hadn't really thought of it like that but now that you mention it, I can see where you're coming from. I still think it's the most logical abstraction though, since if we want to pull that out of that struct then a bunch of call signatures need to change to pass through the features, which I personally like even less. Alternatively, I could wrap it in another struct to keep the metadata separate but I'm not sure that really provides much benefit. Is there a direction you'd prefer?

Reply:
I think keeping with this for now is fine, we'll see how it develops.